src/lib/container.ts

```diff
@@ -21,6 +21,7 @@ export class Container implements IContainer {
             this.parent = <IContainer> options.parent;
             this.defaultLifeTime = options.defaultLifeTime || this.defaultLifeTime;
         }
+        this.register({ token: TContainer, useValue: this });
```

Comment:
You did right! But I'm gonna release that simplified Reflect API which is gonna allow us to inject stuff by just specifying it's type like this `constructor (private container: Container) {}`. Please use Container as the token in this case.